### PR TITLE
66: Integrate DurationParameter as a Python timedelta.

### PR DESCRIPTION
**Pull request by lfse-slafleur**

### CODE REVIEW

File: src/omotes_sdk/workflow_type.py [hunk 1]
```diff
@@ -704,6 +691,114 @@ def to_pb_value(value: ParamsDictValues) -> float:
             )
 
 
+@dataclass(eq=True, frozen=True)
+class DurationParameter(WorkflowParameter):
+    """Define a datetime parameter this SDK supports."""
+
+    type_name: str = "duration"
+    """Parameter type name."""
+    default: Union[timedelta, None] = field(default=None, hash=False, compare=False)
```
Comment (cwang39403) [inline src/omotes_sdk/workflow_type.py:700]: Curious if there is any reason for using `Union[timedelta, None]` instead of `Optional[timedelta]` here?

**Review by cwang39403 — APPROVED** [review]
Looks good, but also got the feeling that it is quite some effort to make Python types to be Protobuf compatible after looking #65 #68, but it is how it is.

**Review by MarkTNO — APPROVED** [review]

File: src/omotes_sdk/workflow_type.py (hunk 1 above)
Comment (lfse-slafleur) [reply src/omotes_sdk/workflow_type.py:700]: Plainly because I copied without looking.. doh! Thanks for the find!

Comment (lfse-slafleur) [reply src/omotes_sdk/workflow_type.py]: Fixed in https://github.com/Project-OMOTES/omotes-sdk-python/pull/68/commits/b3841b3ac320d9013a3f560404463b73a2b12eef

**Comment by lfse-slafleur** [comment]
> Looks good, but also got the feeling that it is quite some effort to make Python types to be Protobuf compatible after looking #65 #68, but it is how it is.

@cwang39403  Yeah we have gathered quite a bit of code but it also does quite a bit for us. While it may look mostly as if it is only to convert from/to protobuf, it actually does a bit more:

- Pack/unpack Python objects into protobuf Struct (which doesn't support int, timedelta or datetime)
- Describe how a parameter should be named and what it does in a workflow definition.
- How to build up this parameter description based on a json file or a protobuf message describing a workflow definition.
- How to dump this parameter to a protobuf message describing a workflow definition.

All in all, it is tightly coupling the workflow definition stuff which is shared from the orchestrator to the SDK, it also helps with validating the params supplied to a job submission and how to parse the config at the worker. It looks deceivingly bloated, but it also appears to fulfill quite a few functionalities. It still feels weird to me as well, but I do not have a better idea currently :P

**Pull request merged**